CONTRIBUTING.md

```diff
@@ -54,31 +57,101 @@ contributed, consistent with the style used by this project. If your
-Transfer](https://github.com/direct-state-transfer) repositories.
+(d) I understand and agree that this project and the contribution
+    are public and that a record of the contribution (including all
+    personal information I submit with it, including my sign-off) is
+    maintained indefinitely and may be redistributed consistent with
+    this project or the open source license(s) involved.
+```
 
-This project also tracks patch provenance and licensing using
-Signed-off-by tags. With the sign-off in a commit message you certify
-that you authored the patch or otherwise have the right to submit it
-under an open source license and you acknowledge that the CLA signed for
-any DST project also applies to this contribution. The procedure is
-simple: just append a line
+With the sign-off in a commit message you certify that you authored the patch
+or otherwise have the right to submit it under an open source license. The
+procedure is simple: To certify above Developer's Certificate of Origin 1.1 for your contribution just append a line
 
-    Signed-off-by: Random J Developer <random@developer.example.org>
+```text
+Signed-off-by: Random J Developer <random@developer.example.org>
+```
 
 to every commit message using your real name or your pseudonym and a valid
 email address.
 
 If you have set your `user.name` and `user.email` git configs you can
-automatically sign the commit by running the git-commit command with the
-`-s` option.  There may be multiple sign-offs if more than one developer
-was involved in authoring the contribution.
+automatically sign the commit by running the git-commit command with the `-s`
+option.  There may be multiple sign-offs if more than one developer was
+involved in authoring the contribution.
+
+For a more detailed description of this procedure, please see
+[SubmittingPatches][] which was extracted from the Linux kernel project, and
+which is stored in an external repository.
+
+### Individual vs. Corporate Contributors
+
+Often employers or academic institution have ownership over code that is
+written in certain circumstances, so please do due diligence to ensure that
+you have the right to submit the code.
+
+If you are a developer who is authorized to contribute to Perun on behalf of
+your employer, then please use your corporate email address in the
+Signed-off-by tag. Otherwise please use a personal email address.
+
+### Maintain Copyright holder / Contributor list
+
+Each contributor is responsible for identifying themselves in the NOTICE
+file, the project's list of copyright holders and authors. Please add
+the respective information corresponding to the Signed-off-by tag as
+part of your first pull request.
+
+If you are a developer who is
+authorized to contribute to perun-node on behalf of your employer, then
+add your company / organization to the list of copyright holders in the
+NOTICE file. As author of a corporate contribution you can also add your
```

Comment:
```suggestion
NOTICE file. As author of a corporate contribution you can ??? also add your
```
Is it optional or mandatory for the author of corporate contribution to use corporate email ?

Reply:
Well, it's optional to add the author.

Reply:
As for your question, that's addressed in the previous section. I'll leave this as is.